src/freenet/node/OpennetManager.java

```diff
@@ -546,6 +589,39 @@ public boolean wantPeer(OpennetPeerNode nodeToAddNow, boolean addAtLRU, boolean
 				return false;
 			}
 		}
+		// Don't connect to peers in excluded countries
+		if (nodeToAddNow != null) {
```

Comment:
I don't know how the geo-lookup code works but I'd blind guess it may not be very cheap given that the 32 bit IPv4 address space is already exhausted.
Thus as an optimization I'd add ```&& excludedCountries.size() > 0``` to this (and take into consideration that this needs to be thread-safe as well as any other accesses to excludedCountries, so I'd probably wrap the whole if() in a locked section)